### PR TITLE
[gtag.js/gtag] add type of get, consent

### PR DESCRIPTION
**Pull request by shuta13**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://developers.google.com/gtagjs/reference/api> , <https://developers.google.com/gtagjs/devguide/consent>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/gtag.js/index.d.ts [hunk 1]
```diff
@@ -5,90 +5,107 @@
...
+        (command: "config", targetId: string, config?: ControlParams | EventParams | CustomParams): void;
+        (command: "set", targetId: string, config: CustomParams): void;
+        (command: "set", config: CustomParams): void;
+        (command: "js", config: Date): void;
+        (
+            command: "event",
+            eventName: EventNames | string,
+            eventParams?: ControlParams | EventParams | CustomParams,
+        ): void;
+        (command: "get", targetId: string, fieldName: FieldNames | string, callback?: (field: string) => any): void;
```
Comment (shuta13) [inline types/gtag.js/index.d.ts:18]: <https://developers.google.com/gtagjs/reference/api#get>

File: types/gtag.js/index.d.ts [hunk 2]
```diff
@@ -5,90 +5,107 @@
...
+        (command: "consent", consentArg: ConsentArg | string, consentParams: ConsentParams): void;
```
Comment (shuta13) [inline types/gtag.js/index.d.ts:19]: <https://developers.google.com/gtagjs/devguide/consent>

File: types/gtag.js/index.d.ts [hunk 3]
```diff
@@ -91,4 +98,14 @@ declare namespace Gtag {
     id?: string;
     name?: string;
   }
+
+  type FieldNames = 'client_id' | 'session_id' | 'gclid';
+
+  type ConsentArg = 'default' | 'update';
+  interface ConsentParams {
+    ad_storage?: 'granted' | 'denied';
```
Comment (rdalmas) [inline types/gtag.js/index.d.ts:106]: According to the API docs, possible values here are 'allowed' and 'denied' for 'ad_storage': https://developers.google.com/gtagjs/reference/api#consent

**Comment by rdalmas** [comment]
I've tried to specify for example a developer Id using gtag and I'm not able. Example would be this:

`gtag('set', 'developer_id', true)`

I think it's missing that signature.

Apart from that, I think it looks good.

**Review by orta — CHANGES_REQUESTED** [review]
Just confirming @rdalmas' feedback

File: types/gtag.js/index.d.ts [hunk 4]
```diff
@@ -91,4 +98,14 @@ declare namespace Gtag {
     id?: string;
     name?: string;
   }
+
+  type FieldNames = 'client_id' | 'session_id' | 'gclid';
+
+  type ConsentArg = 'default' | 'update';
+  interface ConsentParams {
+    ad_storage?: 'allowed' | 'denied';
```
Comment (shuta13) [inline types/gtag.js/index.d.ts:106]: Fixed : https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52148#discussion_r612526045

File: types/gtag.js/index.d.ts [hunk 5]
```diff
@@ -7,10 +7,16 @@ declare var gtag: Gtag.Gtag;
 declare namespace Gtag {
   interface Gtag {
     (command: 'config', targetId: string, config?: ControlParams | EventParams | CustomParams): void;
-    (command: 'set', targetId: string, config: CustomParams): void;
+    (command: 'set', targetId: string, config: CustomParams | boolean): void;
```
Comment (shuta13) [inline types/gtag.js/index.d.ts:10]: Fixed the following points @rdalmas made : https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52148#issuecomment-819631093

> I've tried to specify for example a developer Id using gtag and I'm not able. Example would be this:
> gtag('set', 'developer_id', true)

**Review by shuta13 — COMMENTED** [review]
@rdarida @orta 
Sorry for the late reply. Thank you for your review !
Based on the review content, I have revised the following content. Please check it :)

File: types/gtag.js/index.d.ts (hunk 3 above)
Comment (KasperAndersson) [reply types/gtag.js/index.d.ts]: No, it's not. It says `'granted' | 'denied'` in the [docs](https://developers.google.com/gtagjs/reference/api#consent).

File: types/gtag.js/index.d.ts (hunk 4 above)
Comment (KasperAndersson) [reply types/gtag.js/index.d.ts:106]: Should be updated - https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52148#discussion_r635145966

**Comment by KasperAndersson** [comment]
@rdarida @orta Status?

cc @shuta13

**Comment by sandersn** [comment]
@shuta13 It looks like the remaining question is whether `ConsentParams.ad_storage` has the type `'allowed' | 'denied'` or `'granted' | 'denied'`. Both @KasperAndersson and @rdalmas linked to the same page, where it looks to me that it's `'granted' | 'denied'`

**Comment by shuta13** [comment]
@KasperAndersson @sandersn cc @rdarida @orta 

I did not check the docs enough...

I've fixed it as you suggested. Please check https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52148/commits/5e241249acc0d52d077a62b27c14e44ea270f77e .

**Review by KasperAndersson — APPROVED** [review]

**Comment by KasperAndersson** [comment]
> @KasperAndersson @sandersn cc @rdarida @orta
> 
> I did not check the docs enough...
> 
> I've fixed it as you suggested. Please check [5e24124](https://github.com/DefinitelyTyped/DefinitelyTyped/commit/5e241249acc0d52d077a62b27c14e44ea270f77e) .

You had it right the first time ;)

**Pull request merged**